extensions/incubator/src/test/java/io/opentelemetry/extension/incubator/trace/ExtendedTracerTest.java

```diff
@@ -6,112 +6,225 @@
+                                            OpenTelemetryAssertions.satisfies(
+                                                SemanticAttributes.EXCEPTION_STACKTRACE,
+                                                string ->
+                                                    string.contains(
+                                                        "java.lang.IllegalStateException: ex")),
+                                            OpenTelemetryAssertions.equalTo(
+                                                SemanticAttributes.EXCEPTION_MESSAGE, "ex")))),
+            trace -> trace.hasSpansSatisfyingExactly(a -> a.hasName("another test")));
   }
 
   @Test
-  void runRunnable_throws() {
-    assertThatThrownBy(
-            () ->
-                ExtendedTracer.create(tracer)
-                    .run(
-                        "throwingRunnable",
-                        () -> {
-                          Span.current().setAttribute("one", 1);
-                          throw new RuntimeException("failed");
-                        }))
-        .isInstanceOf(RuntimeException.class);
+  void propagation() {
+    extendedTracer.run(
+        "parent",
+        () -> {
+          ContextPropagators propagators = otelTesting.getOpenTelemetry().getPropagators();
+          Map<String, String> propagationHeaders =
+              ExtendedContextPropagators.getTextMapPropagationContext(propagators);
+          assertThat(propagationHeaders).hasSize(1).containsKey("traceparent");
+
+          // make sure the parent span is not stored in a thread local any more
+          Span invalid = Span.getInvalid();
+          //noinspection unused
+          try (Scope unused = invalid.makeCurrent()) {
+            extendedTracer
+                .spanBuilder("child")
+                .setSpanKind(SpanKind.SERVER)
+                .extractContext(propagators, propagationHeaders)
+                .setAttribute(
+                    "key",
+                    "value") // any method can be called here on the span (and we increase the test
+                // coverage)
+                .setAttribute("key2", 0)
+                .setAttribute("key3", 0.0)
+                .setAttribute("key4", false)
+                .setAttribute(SemanticAttributes.CLIENT_PORT, 1234L)
+                .addLink(invalid.getSpanContext())
+                .addLink(invalid.getSpanContext(), Attributes.empty())
+                .setParent(
+                    Context.current()) // this has no effect, because extractContext() was called
+                // before
+                .setNoParent() // also no effect
+                .setAllAttributes(Attributes.empty())
+                .setStartTimestamp(0, java.util.concurrent.TimeUnit.NANOSECONDS)
+                .setStartTimestamp(Instant.MIN)
+                .run(() -> {});
+          }
+        });
 
     otelTesting
         .assertTraces()
+        .hasSize(1)
```

Comment:
I believe this is redundant. `hasTracesSatisfyingExactly` should assert that size is equal to the number of consumer arguments.